assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetailPage.jsx

```diff
@@ -0,0 +1,112 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import { getCatalog } from '@state/selectors/catalog';
+import { getLastExecution } from '@state/selectors/lastExecutions';
+import { getCluster, getClusterHostNames } from '@state/selectors/cluster';
+import { updateCatalog } from '@state/actions/catalog';
+import {
+  updateLastExecution,
+  executionRequested,
+} from '@state/actions/lastExecutions';
+import {
+  RUNNING_EXECUTION_STATE,
+  REQUESTED_EXECUTION_STATE,
+  RUNNING_STATES,
+} from '@state/lastExecutions';
+import ExecutionContainer from '@components/ExecutionResults/ExecutionContainer';
+import ResultsContainer from '@components/ExecutionResults/ResultsContainer';
+import CheckResultDetail from './CheckResultDetail';
+import { getCheckDescription } from '../checksUtils';
+import CheckDetailHeader from './CheckDetailHeader';
+
+function CheckResultDetailPage() {
```

Comment:
This component is becoming really complex I think.
Too many loading stages.
I know that you did it this way because the page should work standalone, but it adds a ton of complexity, and the user flow is not really good as it needs to query wanda many times.

I don't see tests neither here, and I guess the previous comment is the reason.
Maybe @dottorblaster has some idea on this. At the end, the goal is to have this as pure as possible.

Reply:
I am aware, thanks for pointing out.

As commented in the PR description this `CheckResultDetailPage` has basically the same needs in terms of data and loading behavior as `ExecutionResultsPage`.
They do the same data loading and differ in what they render, and what they render aims to be pure (we're following Container/Presentation pattern where the container - pages - interacts with state and passes data down the tree of pure components, right?)

Yes, we can improve the thing by using react contexts/HOCs, however I think that to be out of scope for what we're trying to achieve here. Don't get me wrong I 'd like to address that as well... by iterating over tho.

About the test you're right `CheckResultDetailPage` is not tested similarly to `ExecutionResultsPage` for the same reasons.

I am open to suggestions.

Reply:
Here's the change about the selector https://github.com/trento-project/web/pull/1248/commits/248e2728b260298092644e1f307284410f1e31f5
I wouldn't go further than that, PR is already growing too much :smile: